### PR TITLE
feat: Add tracking for node errors and update node graph

### PR DESCRIPTION
**Pull request by mutdmour**
## Summary

1. Add runs and items_count to node graph
2. Add tracking for node errors for manual runs

## Review / Merge checklist

- [x] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [x] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/workflow/test/TelemetryHelpers.test.ts [hunk 1]
```diff
@@ -780,6 +780,392 @@ describe('generateNodesGraph', () => {
 
 		expect(() => generateNodesGraph(workflow, nodeTypes)).not.toThrow();
 	});
+
+	test.only('should add run and items count', () => {
+		const workflow: IWorkflowBase = {
```
Comment (RicardoE105) [inline packages/workflow/test/TelemetryHelpers.test.ts:785]: Should we move the JSON to a helper? The helper will create this whole json and override any parameter provided by default. Same for RunData. Will make much easier to read the tests as half of the file is just JSON

File: packages/editor-ui/src/stores/workflows.store.spec.ts [hunk 2]
```diff
@@ -441,4 +451,188 @@ describe('useWorkflowsStore', () => {
 			expect(uiStore.stateIsDirty).toBe(true);
 		});
 	});
+
+	describe('addNodeExecutionData', () => {
+		const executionReponse: IExecutionResponse = {
+			id: '1',
```
Comment (RicardoE105) [inline packages/editor-ui/src/stores/workflows.store.spec.ts:457]: Same as below.

File: packages/workflow/test/TelemetryHelpers.test.ts [hunk 3]
```diff
@@ -780,6 +780,392 @@ describe('generateNodesGraph', () => {
 
 		expect(() => generateNodesGraph(workflow, nodeTypes)).not.toThrow();
 	});
+
+	test.only('should add run and items count', () => {
```
Comment (RicardoE105) [inline packages/workflow/test/TelemetryHelpers.test.ts:784]: Remove only

File: packages/editor-ui/src/stores/workflows.store.spec.ts [hunk 4]
```diff
@@ -8,31 +8,41 @@ import {
...
 }));
 
+const track = vi.fn();
+vi.mock('@/composables/useTelemetry', () => ({
+	useTelemetry: () => ({ track }),
+}));
+
 describe('useWorkflowsStore', () => {
 	let workflowsStore: ReturnType<typeof useWorkflowsStore>;
 	let uiStore: ReturnType<typeof useUIStore>;
```
Comment (RicardoE105) [inline packages/editor-ui/src/stores/workflows.store.spec.ts:39]: `let nodeTypesStore: ReturnType<typeof useNodeTypesStore>;`

File: packages/workflow/src/TelemetryHelpers.ts [hunk 5]
```diff
@@ -200,6 +203,24 @@ export function generateNodesGraph(
 			position: node.position,
 		};
 
+		if (runData?.[node.name]) {
+			const runs = runData[node.name] ?? [];
+			nodeItem.runs = runs.length;
+
+			nodeItem.items_total = runs.reduce((total, run) => {
```
Comment (RicardoE105) [inline packages/workflow/src/TelemetryHelpers.ts:210]: We have 365 lines of code in this function. Should we move this part to a helper so we do not make matters worst? Maybe something like:

`
	nodeItem.items_total = getNumberOfItemsInRun(runs)
`

**Review by RicardoE105 — CHANGES_REQUESTED** [review]

**Review by RicardoE105 — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.64.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.64.0)

**Pull request merged**